electron/main/llm/models/OpenAI.ts

```diff
@@ -1,248 +1,59 @@
-    return this.modelConfig.contextLength || 0;
-  }
-
   public abort(): void {
-    this.abortStreaming = true;
+    throw new Error("Abort not yet implemented.");
   }
 
-  async streamingPrompt(
-    prompt: string,
-    sendFunctionImplementer: ISendFunctionImplementer,
-    generationParams?: LLMGenerationParameters,
-    ignoreChatHistory?: boolean
-  ): Promise<string> {
-    if (!this.isModelLoaded()) {
-      throw new Error("Model not initialized");
-    }
-    this.abortStreaming = false;
-
-    if (ignoreChatHistory) {
-      this.messageHistory = [];
-    }
-
-    // Add the user's prompt to the message history
-    this.messageHistory.push({
-      role: "user",
-      content: prompt,
-      messageType: "success",
+  async streamingResponse(
+    modelName: string,
+    modelConfig: OpenAILLMConfig,
+    messageHistory: ChatCompletionMessageParam[],
+    handleChunk: (chunk: ChatCompletionChunk) => void,
+    generationParams?: LLMGenerationParameters
+  ): Promise<void> {
+    const openai = new OpenAI({
+      apiKey: modelConfig.apiKey,
+      baseURL: modelConfig.apiURL,
+      fetch: customFetchUsingElectronNetStreaming,
+    });
+    // const tokenEncoding = this.getTokenizer(modelName);
+
+    const stream = await openai.chat.completions.create({
+      model: modelName,
+      messages: messageHistory,
+      stream: true,
+      max_tokens: generationParams?.maxTokens,
+      temperature: generationParams?.temperature,
     });
 
-    try {
-      const openAIMessages = this.messageHistory.map((msg) => ({
-        role: msg.role,
-        content: msg.content,
-      })) as OpenAIMessage[];
-      const stream = await this.openai.chat.completions.create({
-        model: this.modelName,
-        messages: openAIMessages,
-        stream: true,
-        max_tokens: generationParams?.maxTokens,
-        temperature: generationParams?.temperature,
-      });
-
-      let result = "";
-      for await (const chunk of stream) {
-        if (this.abortStreaming) {
-          break; // Exit the loop if the flag is set
-        }
-        const content = chunk.choices[0]?.delta?.content || "";
-        result += content;
-
-        // Update the message history with the response
-        this.messageHistory.push({
-          role: "assistant",
-          content,
-          messageType: "success",
-        });
-
-        sendFunctionImplementer.send("tokenStream", {
-          messageType: "success",
-          content,
-        });
-      }
-
-      return result;
-    } catch (error) {
-      console.error("Error during OpenAI streaming session:", error);
-      sendFunctionImplementer.send("tokenStream", {
-        messageType: "error",
-        content: "Error during OpenAI streaming session: " + error + "\n",
-      });
-      return "error";
+    for await (const chunk of stream) {
+      handleChunk(chunk);
     }
   }
 }
-
-export const customFetchUsingElectronNetStreaming = async (
```

Comment:
Extracted out to manage streaming of data

Reply:
This function is to override node fetch defaults to use electron's net module which defaults to using the user's system proxy. node fetch does not use the system proxy for some reason...